src/providers/workers-ai/chatComplete.ts

```diff
@@ -0,0 +1,189 @@
+      transform: (params: Params) => {
+        let messages: Message[] = [];
+        // Transform the chat messages into a simple prompt
+        if (!!params.messages) {
+          params.messages.forEach((msg) => {
+            if (
+              msg.content &&
+              typeof msg.content === 'object' &&
+              msg.content.length
+            ) {
+              const transformedMessage: Record<string, any> = {
+                role: msg.role,
+                content: [],
+              };
+              msg.content.forEach((item) => {
+                if (item.type === 'text') {
+                  transformedMessage.content.push({
+                    type: item.type,
+                    text: item.text,
+                  });
+                } else if (
+                  item.type === 'image_url' &&
+                  item.image_url &&
+                  item.image_url.url
+                ) {
+                  const parts = item.image_url.url.split(';');
+                  if (parts.length === 2) {
+                    const base64ImageParts = parts[1].split(',');
+                    const base64Image = base64ImageParts[1];
+                    const mediaTypeParts = parts[0].split(':');
+                    if (mediaTypeParts.length === 2 && base64Image) {
+                      const mediaType = mediaTypeParts[1];
+                      transformedMessage.content.push({
+                        type: 'image',
+                        source: {
+                          type: 'base64',
+                          media_type: mediaType,
+                          data: base64Image,
+                        },
+                      });
+                    }
+                  }
+                }
+              });
+              messages.push(transformedMessage as Message);
+            } else {
+              messages.push({
+                role: msg.role,
+                content: msg.content,
+              });
+            }
+          });
+        }
+
+        return messages;
+      },
+    },
+  ],
+  stream: {
+    param: 'stream',
+    default: false,
+  },
+};
+
+export interface WorkersAiErrorObject {
+  type: string;
+  message: string;
+}
+
+interface WorkersAiErrorResponse {
+  type: string;
+  error: WorkersAiErrorObject;
+}
+
+interface WorkersAiChatCompleteResponse {
+  result: {
+    response: string;
+  };
+  success: boolean;
+  errors: string[];
+  messages: string[];
+}
+
+interface WorkersAiChatCompleteStreamResponse {
+  response: string;
+  p?: string;
+}
+
+// TODO: cloudflare do not return the usage
+// TODO: return the model
+export const WorkersAiChatCompleteResponseTransform: (
+  response: WorkersAiChatCompleteResponse | WorkersAiErrorResponse,
+  responseStatus: number
+) => ChatCompletionResponse | ErrorResponse = (response, responseStatus) => {
+  if (responseStatus !== 200 && 'errors' in response) {
+    return {
+      error: {
+        message: response.errors?.join(','),
```

Comment:
Cloudflare sends errors as an array of object which has code and message in it. One thing that we can do is to just join the message fields and decorate it with comma.

Example CF errors array:
```
"errors": [
        {
            "code": 7000,
            "message": "No route for that URI"
        }
    ],
```

The current implementation show [object object] because the whole object is joined.

Reply:
@VisargD i handled the errors in https://github.com/Portkey-AI/gateway/pull/269/commits/cad4083d1eb05ac98c6cfdd84b76d1bae52c22f1, thanks for the pointers 👍

Reply:
Here is an error sample : 

```
{
    "error": {
        "message": "workers-ai error: Error 7000:No route for that URI",
        "type": null,
        "param": null,
        "code": null
    },
    "provider": "workers-ai"
}
```